assets/js/product-category-block.js

```diff
@@ -191,6 +214,17 @@ export default class ProductByCategoryBlock extends Component {
 		const { setAttributes } = this.props;
 		const { columns, align, editMode } = this.props.attributes;
 		const { loaded, products } = this.state;
+		const classes = [ 'wc-block-products-category' ];
+		if ( columns ) {
+			classes.push( `cols-${ columns }` );
+		}
+		if ( ! products.length ) {
```

Comment:
While entirely unlikely it will ever be undefined, should we check for products first? `if ( products && ! products.length )`